check again check

ranhershko opened the pull request:
check again check

ranhershko committed:
Add python HW1 & HW2

ranhershko committed:
Delete weather_api.key

ranhershko committed:
Create Elad-code-review

ranhershko committed:
Merge pull request #1 from ranhershko/code-review

Create Elad-code-review

ranhershko committed:
Finished code review

ranhershko committed:
Update age_to_bucket_parser.py

ShimiTaNaka commented on json_2_yaml/age_to_bucket_parser.py:
```diff
@@ -0,0 +1,40 @@
+
+    current_people_list = people_list
+    for age_border in range(len(sorted_ages_borders) - 1):
+        print(sorted_ages_borders[age_border], "-", sorted_ages_borders[age_border + NEXT_AGE_BORDER], ":", sep="")
```
I would strongly recommend not writing serialised files (json, yaml, toml etc) as plain text, and instead use native functions (`yaml.dump()`) and pass inside a dictionary

ShimiTaNaka commented on weather/my_weather.py:
```diff
@@ -0,0 +1,77 @@
+    print(f"The weather in {city} today {temp} {'Fahrenheit' if temp_type == 'f' else 'Celsius'}")
+
+
+def print_random_city_weather():
```
FYI
You are risking an endless loop here since `print_random_city_weather` calls `print_city_weather`, which, in some cases, calls `print_random_city_weather` again
Better splitting of functions can easily solve it

ShimiTaNaka commented on weather/my_weather.py:
```diff
@@ -0,0 +1,77 @@
+
+def print_city_weather(cities_name):
+  api_key = get_api_key()
+  temp_type = "f" if args.f else "m"
```
The variable name does not represent what it stands for

ShimiTaNaka commented on weather/my_weather.py:
```diff
@@ -0,0 +1,77 @@
+        msg = f"\nERROR: Site Api request error answer"
+        print(msg)
+        exit(615)
+    temp = json_out["current"]["temperature"]
```
The variable name does not represent what it stands for

ranhershko committed:
Add setup.py => setuptools/setup module
	Rename weather.py file
	Use click module instead

ranhershko committed:
Update weather.py

variables names fixed

ranhershko committed:
Update yaml output using yaml.dump

ranhershko committed:
Merge branch 'master' of https://github.com/ranhershko/opsschool5-coding

ranhershko committed:
change to meaningble variables name

ranhershko committed:
change to meaningble variables name

ShimiTaNaka commented on weather/weather.py:
```diff
@@ -0,0 +1,50 @@
+def cli(temperature, city, token):
+    cities = city.replace(';', ',').split(',')
+    for_supplied_cities = [city_exists for city_exists in cities if city_exists]
+    in_temperature_unit = temperature
```
Redundant variable? what is it used for?

ranhershko replied:
I don't understand why  it redundant ?

ranhershko replied:
city variable is not redundant as the cities,for_supplied_cities are manipulation of the variable city and not using the same value 
Also I created the for_supplied_cities variable to make code more  readable when I call the function

ranhershko replied:
print_city_weather(for_supplied_cities, in_temperature_unit, using_api_token)

ShimiTaNaka commented on weather/weather.py:
```diff
@@ -0,0 +1,50 @@
+    cities = city.replace(';', ',').split(',')
+    for_supplied_cities = [city_exists for city_exists in cities if city_exists]
+    in_temperature_unit = temperature
+    using_api_token = token
```
Yet another redundant variable?

ranhershko replied:
The token variable is the variable that I present to the user and I found its more right naming in this position.  
I created the using_api_token variable to make code more readable when I call the function

ranhershko replied:
print_city_weather(for_supplied_cities, in_temperature_unit, using_api_token)

ShimiTaNaka commented on weather/weather.py:
```diff
@@ -0,0 +1,50 @@
+def print_city_weather(cities, temperature, token):
+    temperature_unit = "m" if temperature == 'Celsius' else "f"
+    for city in cities:
+        url = f"http://api.weatherstack.com/current?access_key={token}&query={city}&units={temperature_unit}"
```
FYI,
Per documentation, you can do it all in a single API call, see https://weatherstack.com/documentation#query_parameter under multiple locations

ranhershko replied:
I know, to use it you need a token that cost money

ShimiTaNaka commented on weather/weather.py:
```diff
@@ -0,0 +1,50 @@
+                print(msg)
+                exit(615)
+        current_city_temperature = json_out["current"]["temperature"]
+        print(f"The weather in {json_out['request']['query']} today {current_city_temperature} \
```
Any reason fir the inconsistency?
for the temperature you assign a temp variable, but for the query you fetch it directly from the JSON?

ranhershko replied:
The reason is:
The current_city_temperature temperary variable is set from the url request and used for display in tne print output in the end

The city variable been used in url request but when I print the output I want to display the city and the country values that exist only in json_out['request']['query'] key value

ShimiTaNaka commented on weather/weather.py:
```diff
@@ -0,0 +1,50 @@
+                exit(615)
+        current_city_temperature = json_out["current"]["temperature"]
+        print(f"The weather in {json_out['request']['query']} today {current_city_temperature} \
+{'Fahrenheit' if temperature_unit == 'f' else 'Celsius'}")
```
FYI
You already have this information in the `temperature` variable, no real need for the `if` here

ranhershko replied:
That is right, I will replace it

ShimiTaNaka reviewed (commented):
Overall, good job, please see comments

ranhershko committed:
Update weather.py

ranhershko reviewed (commented):
That's is my comments

ranhershko committed:
Update weather.py

ranhershko committed:
Update weather.py

ranhershko committed:
Modified: weather/weather.py
1. make variables name more meaningful
2. Remove unneeded temporary variables use

ranhershko committed:
Removed unneeded temporary variable
